crates/composer/src/main.rs

```diff
@@ -0,0 +1,37 @@
+use composer::api::Event;
+use composer::api::DEFAULT_SERVER_ADDRESS;
+use eyre::Result;
+use rodio::{source::Source, Decoder, OutputStream};
+use std::fs::File;
+use std::io::BufReader;
+use std::net::UdpSocket;
+
+fn main() -> Result<()> {
+    color_eyre::install()?;
+
+    let socket = UdpSocket::bind(DEFAULT_SERVER_ADDRESS)?;
+    println!("Listening on {}", socket.local_addr()?);
+
+    let (_stream, stream_handle) = OutputStream::try_default()?;
+
+    loop {
+        if let Err(err) = {
+            // Should be more than enough.
+            let mut buf = [0; 1000];
+            let (number_of_bytes, _) = socket.recv_from(&mut buf)?;
+
+            let event: Event = bincode::deserialize(&buf)?;
```

Comment:
> I think I can replay-attack this by sending a 0-size UDB packet. I think it will trigger the current code to replay the previous last event to be played again.

How is this possible if the buffer is reset to all zeros on every iteration of the loop? The buffer binding is inside of the loop, so it should be zero'd every pass.

Reply:
Ah @PabloMansanet true - for some reason I assumed the buffer is reused across requests, but it is indeed zeroed every iteration.

Reply:
@PabloMansanet yup, and if I'm not mistaken, since the buffer is stack-allocated, re-declaring it with every iteration has basically the same cost as manually zeroing it out so this is reasonably effective defense against replay attacks.